### ISSUE TITLE
Issue building GCAM 5.3 in Windows

### ISSUE THREAD

**Issue by emuwren2003** [JGCRI/gcam-core#150]
Hello:  

I am using Visual Studio 2019 to build GCAM 5.3.  I have followed the steps in: https://jgcri.github.io/gcam-doc/gcam-build.html.  However, I get this error: Cannot open 'libboost_system-vc142-mt-x64-1_73.lib'.  Does anyone know how to resolve this?

**Comment by pralitp** [comment]
This indicates that either the boost `system` library did not build correctly (and probably `filesystem` as well).  Or there was some difference between the version / configuration of Visual Studio you used to build boost and what you are now using to build GCAM.

You you check what files you have in your `libs/boost-lib/stage/lib` .

If there is nothing there then that indicates there was an error when building them in the first place.

If there is then check the exactly naming.  It will likely indicate which version Visual Studio was used.

**Comment by Silviameteoro** [comment]
Hi @pralitp , could you help since I am having a similar issue? In my case, I am using Visual Studio 2022 and I am trying to build GCAM v5.4. I have received the error message: fatal error LNK1104: cannot open file 'libboost_filesystem-vc142-mt-x64-1_70.lib' .

I have followed the suggestion and I have checked my folder libs/boost-lib/stage/lib, which is empty. Not sure what gave wrong. 

To try to build the boost library I have used the following commands using the Develop Command Prompt for VS 2022:
1) bootstrap vc141   -> I have received a message that includes "Bootstrapping is done". 
2) b2 --with-system --with-filesystem address-model=64 stage  -> I have received a bunch of messages that seem to indicate an issue (I am not quite understanding admittedly). I believe this is the source of my problem.

I am attaching the print screens of the messages that I have received after 2) . Could you advise me on what to do? Thanks a lot  in advance.
![build1](https://user-images.githubusercontent.com/24659743/157532756-3c0cd2b8-0634-430f-a6d2-0f3e3ec9f339.png)
![build2](https://user-images.githubusercontent.com/24659743/157532778-297a9df3-723f-4efb-b89f-a1d63b47c80d.png)
.

**Comment by pralitp** [comment]
@Silviameteoro, I am not sure as I haven't used Visual Studio 2022 which I gather came out just a few weeks ago.

Thus I can just offer a few thoughts at the moment:

The error `cannot open file 'libboost_filesystem-vc142-mt-x64-1_70.lib'` seems to indicate it wants `vc142`.  Any reason you used `vc141` instead in in the following?
> bootstrap vc141 -> I have received a message that includes "Bootstrapping is done".

That being said I gave a quick google and found:
https://stackoverflow.com/questions/68935077/how-to-compile-boost-library-with-visual-studio-2022-preview-3

Which seems to insinuative Visual Studio 2022 uses the toolset `vc143` now.  There are no accepted answers.  One of the replies suggests the latest version of Boost 1.78 now supports it.  However, in the release notes for 1.78 it only lists support up to vc142.

In summary I would try:
1. When building boost try to use the vc142 toolset instead of vc141 and see if that works.
2. Next, try upgrading your boost to 1.78 and try vc142; if not try vc143 in case it is actually supported.
3. If all else fails you may have to downgrade back to Visual Studio 2019.

**Comment by Silviameteoro** [comment]
Thanks for the reply @pralitp. 

I have tried to follow your suggestions. I ended up downgrading to Visual Studio 2019, however I am still having problems. 

I am suspecting about 2 things: 
1) I am not entirely sure if my boost-lib library compiled correctly;  
2) I am not sure if paths are correctly set.

I am trying to debug below these 2 points:
1) boost-lib

To try to build the boost library I have used the same commands as before using the Develop Command Prompt for VS 2019:

bootstrap vc141 
b2 --with-system --with-filesystem address-model=64 stage  -> Differently from the first time, now I have files in libs/boost-lib/stage/lib (figure Boost_lib_stage_lib_content). But I am not sure if those are all files that are supposed to be there or if there are missing components. The messages that were raised after this command, make suspicious that things are not still OK with the boost-lib (figures Boost-lib_1 to 4).

What do you think about this? Thanks.

![Boost_lib_stage_lib_content](https://user-images.githubusercontent.com/24659743/157768187-8a32d986-a50a-4988-ad32-7211e25421ef.png)
![Boost_lib_4](https://user-images.githubusercontent.com/24659743/157768979-4cf1d9bb-1a22-42aa-8c2b-97eb2839796f.png)
![Boost_lib_1](https://user-images.githubusercontent.com/24659743/157768980-a2fb0be1-ce37-4ce7-bc80-792e3aad6f5d.png)
![Boost_lib_2](https://user-images.githubusercontent.com/24659743/157768982-e10bd67f-700d-48bb-b81c-a98045c5a790.png)
![Boost_lib_3](https://user-images.githubusercontent.com/24659743/157768983-0a0cbcab-20b4-414c-a261-e8529973d0ba.png)

**Comment by Silviameteoro** [comment]
About paths, what I have for hector-lib and the objects-main are in the figures below. Do you think there are missing components or things to adjust? Thank you very muc
![objects_main_path](https://user-images.githubusercontent.com/24659743/157770425-89514a37-ad03-4567-87e0-0388b1f948fa.png)
![hector-lib_path](https://user-images.githubusercontent.com/24659743/157770427-3cc6931e-e31c-428c-9b81-83feea6ce38e.png)
![objects_main_linker_path](https://user-images.githubusercontent.com/24659743/157770428-21d07f3a-ad5c-429a-933d-4f201e183e62.png)
h

**Comment by pralitp** [comment]
Looks ok to me.  What error do you get when you are compiling GCAM?

I suspect it still wants `vc142`.  Certainly the windows builds I've done using Visual Studio 2019 do by default.

Could you provide some insight about why are you explicitly asking for `vc141` in:
```
bootstrap vc141
```

Was there some error when you tried just `bootstrap` or `bootstrap vc142`?
